### PR TITLE
Add codegen attribute tags

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Fixes: https://github.com/Azure/autorest.csharp/issues/4385
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Input/Source/CodeGenAttributes.cs [hunk 1]
```diff
@@ -15,25 +15,24 @@ public class CodeGenAttributes
...
-            CodeGenMemberAttribute = GetSymbol(compilation, typeof(CodeGenMemberAttribute));
-            CodeGenTypeAttribute = GetSymbol(compilation, typeof(CodeGenTypeAttribute));
-            CodeGenModelAttribute = GetSymbol(compilation, typeof(CodeGenModelAttribute));
-            CodeGenClientAttribute = GetSymbol(compilation, typeof(CodeGenClientAttribute));
-            CodeGenSerializationAttribute = GetSymbol(compilation, typeof(CodeGenSerializationAttribute));
+            CodeGenMemberAttributeName = "CodeGenMemberAttribute";
+            CodeGenTypeAttributeName = "CodeGenTypeAttribute";
+            CodeGenModelAttributeName = "CodeGenModelAttribute";
+            CodeGenClientAttributeName = "CodeGenClientAttribute";
+            CodeGenSerializationAttributeName = "CodeGenSerializationAttribute";
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Common/Input/Source/CodeGenAttributes.cs:22]: we could make these as const strings as well

**Review by ArcturusZhang — COMMENTED** [review]
As a general comment, I kind of suggest we unify the way of introducing customization code via attributes in branded and non-branded libraries.
The generator could generate those attributes, in this way we do not have to write the attributes (some of them contain quite a few properties).
If we choose this way, we might want to generate those attributes in a fixed namespace (such as something like `Generator.Internal`) instead of in the namespace of the library (because of the [apicomat settings](https://github.com/Azure/azure-sdk-for-net/blob/00af2ba6303ae616ba247cba0415151fe351ddb4/eng/ApiListing.exclude-attributes.txt#L6) in azure-sdk-for-net, as far as we know, this does not support wild cards)

**Comment by m-nash** [comment]
@ArcturusZhang I do think unifying is what we want to achieve, however I think in mgc these will be public attributes instead of making customers create their own copies of them.

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang I do think unifying is what we want to achieve, however I think in mgc these will be public attributes instead of making customers create their own copies of them.

yeah making them public is a way, but I think if we let the generator to generate those attribute, in this way, we do not have to make those attribute public, which might be valuable since the public API is reduced, and those attributes are useless outside this generated library, therefore I think if possible to keep them internal, we should.

Customers are impossible to use these attributes without a document, therefore I am imagining this way of using them:
The generator will always generate them in the same namespace, but since no classes are using them, they will be removed by post processor (I think we still will have this in MGC, right?), and the customer reads the document to know that I need to add this attribute on the customized code and regenerate, they just write the code, despite the attribute does not exist, after they regenerate, the attribute now has a usage, and will no longer be removed, and everything should work as expected.

What do you think? I think this is a way to do both "not make them public" and "make everything to work properly" considering the fact that every time we add customization code, we will have to regenerate the library again

**Comment by m-nash** [comment]
@ArcturusZhang opened a tracking issue to finalize on this https://github.com/Azure/autorest.csharp/issues/4551

**Review by m-nash — APPROVED** [review]

**Pull request merged**